doc/source/security/vault.rst

```diff
@@ -5,29 +5,210 @@ Information about HashiCorp Vault
+         - `KV Secrets Engine - Version 1 <https://www.vaultproject.io/docs/secrets/kv/kv-v1.html>`_
+
+      Production Hardening
+         https://learn.hashicorp.com/vault/operations/production-hardening
+
+.. admonition:: Related information
+   
+   - :ref:`using-keyring-plugin`
+   - :ref:`rotating-master-key`
+
+.. _vault/custom-server.setting-up:
+
+Setting up a custom Vault server
+================================================================================
+
+The ``keyring_vault`` plugin interacts with a |vault-server| to save and retrieve
+encryption keys. On |vault-server|, all keys are stored securely and under
+strict control. This section provides a general overview of how to set up custom
+server that you can experiment with.
+
+.. _vault/custom-server.setting-up/installing:
+
+Installing the ``vault`` command
+--------------------------------------------------------------------------------
+
+In order to use |vault-server| features, you need to install the |command.vault|
+provided by |hashicorp-vault|. The installation procedure is straightforward:
+
+1. Download the installation package from the |hashicorp-vault| site and unzip
+   it.
+#. Place the ``vault`` binary to a location in your `PATH`.
+#. Run the ``vault`` command (:ref:`vault/tip/command-not-found`):
+
+   .. code-block:: bash
+
+      $ vault
+
+      Usage: vault <command> [args]
+
+      Common commands:
+         read        Read data and retrieves secrets
+         write       Write data, configuration, and secrets
+         delete      Delete secrets and configuration
+         list        List data or secrets
+         login       Authenticate locally
+         agent       Start a Vault agent
+         server      Start a Vault server
+         status      Print seal and HA status
+         unwrap      Unwrap a wrapped secret
+      ...
+
+.. _vault/tip/command-not-found:
+
+.. tip:: Command not found instead of the usage information
+
+   If you receive a ``command not found` message instead, then ``vault`` is not
+   in your `PATH`. To check if the directory with `vault` is included, run:
+
+   $ echo $PATH
+
+.. seealso:: `Vault documentation: Installing Vault
+	     <https://learn.hashicorp.com/vault/getting-started/install>`_
+   
+.. _vault/custom-server.setting-up/starting:
+
+Starting the Vault server
+--------------------------------------------------------------------------------
+
+There are two options to start the Vault server. In dev mode, you can use it for
+experimentation: 
+
+.. code-block:: bash
+
+   $ vault server -dev
+
+Run this way, |vault| features automatic authentication and in-memory
+storage. |vault-server| is not suitable for production use in this mode.
+
+Alternatively, you can *deploy* |vault-server| in your environment. For
+deployment, you use *hcl* configuration files (custom format similar to JSON
+developed by |hashicorp-vault|) where you can set values of multiple
+|vault-server| options.
+
```

Comment:
line 97, we mention HCL. Should we point out that the following is an example of HCL?

Reply:
we should instead link to the hashicorp documentation on configuration HCL for vault,

Reply:
The paragraph above the example mentions that it HCL is similar to  JSON. The example illustrates that. We are safe to assume that the reader is aware of JSON (taking into account the context.)

Reply:
@Oneiroi Note the seealso block after the example:
  |hashicorp-vault| Documentation:
      - `Deploying Vault server <https://learn.hashicorp.com/vault/getting-started/deploy>`_
      - `Vault server configuration options <https://www.vaultproject.io/docs/configuration>`_
It provides references to the deployment procedure itself and the configuration file and available options